hypothesis-python/src/hypothesis/extra/ghostwriter.py

```diff
@@ -800,13 +880,20 @@ def _write_call(
     which `func` might raise, and catch-and-reject on them... *unless* they're
     subtypes of `except_`, which will be handled in an outer try-except block.
     """
+    params_dd = _get_params(func)
+    if copy_args:
+        potentially_modified = potentially_modified_vars(ast.parse(inspect.getsource(func)).body[0], set(params_dd.keys()))
+    else:
+        potentially_modified = set()
```

Comment:
I think we probably want to deepcopy all arguments unconditionally if `copy_args=True`, and autodetect on `=None`.

The code would be a bit cleaner if we also do a `from copy import deepcopy`, meaning this function would report a `str, set-of-imports` tuple - there are plenty of examples you can see of how that works.